File: backend/app/routers/job.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.schemas.job import JobCreate, JobResponse, PaginatedJobResponse
from app.services.job_service import JobService
from app.core.dependencies import get_current_client, get_current_user, get_current_freelancer

router = APIRouter(prefix="/jobs", tags=["Jobs"])

@router.post("/create", response_model=JobResponse)
def create_job(data: JobCreate, db: Session = Depends(get_db), user = Depends(get_current_client)):
    return JobService.create_job(db, data, user)

@router.get("/", response_model=PaginatedJobResponse)
def list_jobs(db: Session = Depends(get_db),
              user = Depends(get_current_user),
              skip: int = 0,
              limit: int = 50):
    return JobService.list_jobs(db, user, skip, limit)

@router.get("/me")
def get_my_jobs(db: Session = Depends(get_db), user = Depends(get_current_client)):
    return JobService.get_my_jobs(db, user)

@router.get("/{job_id}")
def get_job(job_id: int, db: Session = Depends(get_db), user = Depends(get_current_user)):
    return JobService.get_job(db, job_id)

@router.post("/complete/{job_id}")
def complete_job(job_id: int, db: Session = Depends(get_db), user = Depends(get_current_client)):
    return JobService.complete_job(db, job_id, user)

@router.delete("/{job_id}")
def delete_job(job_id: int, db: Session = Depends(get_db), user = Depends(get_current_client)):
    return JobService.delete_job(db, job_id, user)

File: frontend-clean/src/app/services/job.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class JobService {

  API = `${environment.apiUrl}/jobs/`;

  constructor(private http: HttpClient) {}

  getJobs(skip: number = 0, limit: number = 50) {
    return this.http.get<any>(`${this.API}?skip=${skip}&limit=${limit}`);
  }
  getMyJobs() {
    return this.http.get<any[]>(`${environment.apiUrl}/jobs/me`);
  }
  createJob(data: any) {
  return this.http.post(`${environment.apiUrl}/jobs/create`, data);
  }
  deleteJobAsAdmin(id: number) {
    return this.http.delete(`${environment.apiUrl}/admin/jobs/${id}`);
  }
  completeJob(id: number) {
    return this.http.post(`${environment.apiUrl}/jobs/complete/${id}`, {});
  }
  deleteJobAsClient(id: number) {
    return this.http.delete(`${environment.apiUrl}/jobs/${id}`);
  }
}